src/main/java/com/google/sps/data/Course.java

```diff
@@ -35,4 +56,54 @@ public String getTerm() {
   public String getSchool() {
     return school;
   }
+
+  public Key getTermKey() {
+    return termKey;
+  }
+
+  public void setTermKey(Key key) {
+    termKey = key;
+  }
+
+  @Override
+  public int compareTo(Course course) {
+    return name.compareTo(course.name);
+  }
+
+  public int compare(Course one, Course two) {
+    return one.compareTo(two);
+  }
+
+  private List<Key> findKeys(
+      DatastoreService db,
+      String schoolName,
+      String courseName,
+      String termName,
+      Long units,
+      String profName) {
+    Key schoolKey = findQueryMatch(db, "School", "school-name", schoolName).get(0).getKey();
+    List<Filter> filters = new ArrayList();
+    Filter courseFilter = new FilterPredicate("course-name", FilterOperator.EQUAL, courseName);
+    Filter unitFilter = new FilterPredicate("units", FilterOperator.EQUAL, units);
+    filters.add(courseFilter);
+    filters.add(unitFilter);
+
+    Query courseQuery =
+        new Query("Course").setAncestor(schoolKey).setFilter(CompositeFilterOperator.and(filters));
+    Key courseKey =
+        db.prepare(courseQuery).asList(FetchOptions.Builder.withDefaults()).get(0).getKey();
+    Filter termFilter = new FilterPredicate("term", FilterOperator.EQUAL, termName);
+    Query termQuery = new Query("Term").setAncestor(courseKey).setFilter(termFilter);
+    Entity foundTerm = db.prepare(termQuery).asList(FetchOptions.Builder.withDefaults()).get(0);
+    List<Key> keys = Arrays.asList(foundTerm.getKey(), courseKey);
+    return keys;
+  }
+
+  private List<Entity> findQueryMatch(
+      DatastoreService db, String entityType, String entityProperty, String propertyValue) {
+    Filter filter = new FilterPredicate(entityProperty, FilterOperator.EQUAL, propertyValue);
+    Query q = new Query(entityType).setFilter(filter);
+    List<Entity> result = db.prepare(q).asList(FetchOptions.Builder.withDefaults());
```

Comment:
Seems like this is something I do a lot - I thought it was better style bc the variable name is clear but definitely see how it's useless. Sorry for any annoyance/confusion haha

Reply:
That's not invalid. Particularly in JavaScript that can often be helpful since there's no typing on the return statement to clue you into what's being passed back. 

It just is usually redundant in Java, and often the function name should tell you what it's returning anyways. 
It also is a very slight memory increase (extra variable stored) but so slight it would never be noticeable on benchmarks lol. Just more the principle of the thing.